include/nbl/system/SReadWriteSpinLock.h

```diff
@@ -0,0 +1,98 @@
+#ifndef __NBL_S_READ_WRITE_SPIN_LOCK_H_INCLUDED__
+#define __NBL_S_READ_WRITE_SPIN_LOCK_H_INCLUDED__
+
+#include <atomic>
+#include <thread>
+
+namespace nbl {
+namespace system
+{
+
+namespace impl
+{
+
+    struct SReadWriteSpinLockBase
```

Comment:
ok so I guess this replaces the crap in "FW_Mutex.h" ?

Reply:
hm, didnt thought about that, but yeah, basically yeah

Reply:
wonderful, resolved

Reply:
delete "FW_Mutex.h" and mark as resolved